process/smartContract/builtInFunctions/esdtFreezeWipe.go

```diff
@@ -0,0 +1,102 @@
+package builtInFunctions
+
+import (
+	"bytes"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/vm"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+var _ process.BuiltinFunction = (*esdtFreezeWipe)(nil)
+
+type esdtFreezeWipe struct {
+	marshalizer marshal.Marshalizer
+	keyPrefix   []byte
+	wipe        bool
+	freeze      bool
+}
+
+// NewESDTFreezeWipeFunc returns the esdt freeze/un-freeze/wipe built-in function component
+func NewESDTFreezeWipeFunc(
```

Comment:
I would rename this, so it won't have both freeze and wipe in the name. But I don't have a good enough suggestion right now, so if you also don't have a better idea, it can stay like this

Reply:
yep, I did not have any good idea.